catalog/core/catalog-core-api/src/main/java/ddf/catalog/filter/SimpleFilterDelegate.java

```diff
@@ -0,0 +1,679 @@
+        return defaultOperation(propertyName, literal, literalClass, temporalPropertyOperation);
+    }
+
+    public <S> T xpathOperation(String xpath, S literal, Class<S> literalClass,
+            XPathPropertyOperation xpathPropertyOperation) {
+        return defaultOperation(xpath, literal, literalClass, xpathPropertyOperation);
+    }
+
+    // Logical operators
+    public T and(List<T> operands) {
+        return logicalOperation(operands, LogicalPropertyOperation.AND);
+    }
+
+    public T or(List<T> operands) {
+        return logicalOperation(operands, LogicalPropertyOperation.OR);
+    }
+
+    public T not(T operand) {
+        return logicalOperation(operand, LogicalPropertyOperation.NOT);
+    }
+
+    public T include() {
+        return logicalOperation(null, LogicalPropertyOperation.INCLUDE);
+    }
+
+    public T exclude() {
+        return logicalOperation(null, LogicalPropertyOperation.EXCLUDE);
+    }
+
+    // Comparison operators
+
+    // PropertyIsEqualTo
+
+    public T propertyIsEqualTo(String propertyName, String literal, boolean isCaseSensitive) {
+        return propertyIsEqualTo(propertyName,
+                literal,
+                String.class,
+                ComparisonPropertyOperation.IS_EQUAL_TO);
+    }
+
+    public T propertyIsEqualTo(String propertyName, Date literal) {
+        return propertyIsEqualTo(propertyName,
+                literal,
+                Date.class,
+                ComparisonPropertyOperation.IS_EQUAL_TO);
+    }
+
+    public T propertyIsEqualTo(String propertyName, Date startDate, Date endDate) {
+        return propertyIsEqualTo(propertyName,
+                new DateRange(startDate, endDate),
+                DateRange.class,
+                ComparisonPropertyOperation.IS_EQUAL_TO);
+    }
+
+    public T propertyIsEqualTo(String propertyName, int literal) {
+        return propertyIsEqualTo(propertyName, (Number) literal);
+    }
+
+    public T propertyIsEqualTo(String propertyName, short literal) {
+        return propertyIsEqualTo(propertyName, (Number) literal);
+    }
+
+    public T propertyIsEqualTo(String propertyName, long literal) {
+        return propertyIsEqualTo(propertyName, (Number) literal);
+    }
+
+    public T propertyIsEqualTo(String propertyName, float literal) {
+        return propertyIsEqualTo(propertyName, (Number) literal);
+    }
+
+    public T propertyIsEqualTo(String propertyName, double literal) {
+        return propertyIsEqualTo(propertyName, (Number) literal);
+    }
+
+    public T propertyIsEqualTo(String propertyName, Number literal) {
+        return propertyIsEqualTo(propertyName,
+                literal,
+                Number.class,
+                ComparisonPropertyOperation.IS_EQUAL_TO);
+    }
+
+    public T propertyIsEqualTo(String propertyName, boolean literal) {
+        return propertyIsEqualTo(propertyName,
+                literal,
+                boolean.class,
```

Comment:
You want this to be the class of the primitive boolean and not `Boolean`? With all the numeric primitives, you autoboxed and cast to `Number`.